### ISSUE TITLE
Table Tennis Ball Position Detection

### ISSUE THREAD

**Issue by abhisheks008** [World-of-ML/DL-Simplified#243]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : TT Ball Position Detection</br>
:red_circle: **Aim** : Create a DL model which will detect the ball position in a TT table.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/ketzoomer/table-tennis-ball-position-detection-dataset</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.
**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by rafiya2003** [comment]
I would like to work on this, frankly I don't know much but I will learn and do. I am a newbie(first year UG student). I want to learn and grow

**Comment by abhisheks008** [comment]
Please do not to spam in every single issue. This might cause escalation as you are breaching Code of Conduct and Contribution Guidelines.

@rafiya2003

**Comment by Aspireve** [comment]
Hey Hi @abhisheks008 Could I work on this issue

- Full name : Steve Fernandes
- GitHub Profile Link : https://github.com/Aspireve
- Email ID : steve1818fernandes@gmail.com
- Participant ID (if applicable):
- Approach for this Project : I actually have no current idea about how to do it but I have researched a bit about ball detection on kaggle and got the below links
  - https://www.kaggle.com/code/vishakkbhat/yolov7-explanation-and-implementation-from-scratch
  - https://www.kaggle.com/code/zzy990106/fork-of-fork-of-dfl-notebook-4fold
  - https://www.kaggle.com/code/vbookshelf/basics-of-yolo-v5-balloon-detection
  
  and the following YouTube videos (not for ball detection but related to object detection)
  - https://www.youtube.com/watch?v=fu2tfOV9vbY
 
  I guess I might be able to do it by referring to the code documentation and implementations in these videos
  (Am I going in the right way ? Do share your thoughts)
- What is your participant role? Contributor @ SSOC

**Comment by Veena4512** [comment]
@abhisheks008  I know how to solve this issue. Can you please assign it to me?

**Comment by abhisheks008** [comment]
I like the approach of @Aspireve. Try to implement that in your project. Issue assigned to you @Aspireve

**Comment by Aspireve** [comment]
Hey Thanks for this opportunity!! Ill start working on it!!

**Comment by rafiya2003** [comment]
@abhisheks008 sir, with all due respect. I was just trying to get some issues to work on and had no intention to spam.

**Comment by abhisheks008** [comment]
> @abhisheks008 sir, with all due respect. I was just trying to get some issues to work on and had no intention to spam.

It's okay, no problem.

**Comment by ShaikArshidBanu** [comment]
Hi @abhisheks008 I would like to work on this issue

Full name : Shaik Arshid Banu
GitHub Profile Link : https://github.com/ShaikArshidBanu
Email ID : arshidbanushaik@gmail.com
Participant ID (if applicable):NA

Approach for this Project : 
1. Exploratory Data Analysis (EDA)
Loading the Dataset: Loading and understanding the structure of the dataset.
Data Statistics: Calculate basic statistics about the dataset such as the number of images, distribution of ball positions, and so on.

2. Preprocessing
Image Resizing: Resizing images to a consistent size suitable for the models.
Normalization: Normalizing pixel values to aid in model convergence.
Splitting the Dataset: Dividing the dataset into training and validation sets.
Data Augmentation: Applying transformations like rotation, flipping, and scaling to improve model robustness.

3. Model Building
Implementing the following models:

(i) Transfer Learning with Pre-trained Models:
Using pre-trained models  VGG16, ResNet50 and fine-tune them for ball position detection.

(ii) YOLO (You Only Look Once):
Implementing a YOLO model, which is well-suited for object detection tasks.

(iii) EfficientDet:
A more recent and efficient model for object detection.

4. Comparison and Evaluation
Training and Validation: Training each model and evaluating their performance on the validation set.
Performance Metrics: Using metrics such as Mean Squared Error (MSE) for position regression, precision, recall, and F1-score for detection accuracy.
Model Comparison: Compare the performance metrics of all models to identify the best one.

What is your participant role? (Mention the Open Source program) contributor @ GSSOC'24

**Comment by abhisheks008** [comment]
Assigned to you @ShaikArshidBanu
